runtime/compiler/optimizer/JProfilingBlock.cpp

```diff
@@ -710,9 +710,9 @@ int32_t TR_JProfilingBlock::processCFGForCounting(BlockParents &parent, TR::Bloc
          // if the destination of the edge has a single destination we can count the destination
          // if the destination is a catch block we also just count the destination
          else if (block != to &&
-             (to->getPredecessors().size() == 1 && to->getExceptionPredecessors().size() == 0)
+             ((to->getPredecessors().size() == 1 && to->getExceptionPredecessors().size() == 0)
              || (to->getPredecessors().size() == 0 && to->getExceptionPredecessors().size() == 1)
-             || to->isCatchBlock() || to->isOSRInduceBlock() || to->isOSRCodeBlock())
+             || to->isCatchBlock() || to->isOSRInduceBlock() || to->isOSRCodeBlock()))
             {
```

Comment:
Could I get someone familiar with this code to review this? This is a change in behaviour, but reading the code I think this is what we really wanted to check for? @r30shah/@jdmpapin  perhaps?